master/pkg/searcher/random.go

```diff
@@ -68,13 +68,24 @@ func (s *randomSearch) initialOperations(ctx context) ([]Operation, error) {
 	return ops, nil
 }
 
-func (s *randomSearch) progress(trialProgress map[model.RequestID]model.PartialUnits) float64 {
+func (s *randomSearch) progress(
+	trialProgress map[model.RequestID]model.PartialUnits,
+	trialsClosed map[model.RequestID]bool) float64 {
 	if s.MaxConcurrentTrials() > 0 && s.PendingTrials > s.MaxConcurrentTrials() {
 		panic("pending trials is greater than max_concurrent_trials")
 	}
-	unitsCompleted := sumTrialLengths(trialProgress)
+	unitsCompleted := 0.
+	for k, v := range trialProgress {
+		if closed, ok := trialsClosed[k]; ok {
+			if closed {
+				unitsCompleted += float64(s.MaxLength().Units)
+			}
```

Comment:
existence in this map indicates a trial is closed (it's just a set, but go doesn't have a set), not the bool (this is where it may be better to use `map[...]struct{}{}`. but since we always set it to true, we can use it like below.
```suggestion
		if trialsClosed[k] {
			unitsCompleted += float64(s.MaxLength().Units)
```